alea/test/stream_serializer.cpp

```diff
@@ -0,0 +1,167 @@
+#include <alps/alea/plugin/stream_serializer.hpp>
+
+#include "gtest/gtest.h"
+#include "dataset.hpp"
+
+#include <vector>
+#include <queue>
+
+// Mock class that emulates Boost/HPX serialization archive
+class MockArchive {
+
+public:
+
+    MockArchive() {}
+
+    // Store simple types
+    MockArchive & operator<<(double x) { store_fundamental(x); return *this; }
+    MockArchive & operator<<(long x) { store_fundamental(x); return *this; }
+    MockArchive & operator<<(unsigned long x) { store_fundamental(x); return *this; }
+    MockArchive & operator<<(std::complex<double> x)
+    {
+        *this << x.real() << x.imag();
+        return *this;
+    }
+
+    // Store complex_op<T>
+    template<typename T>
+    MockArchive & operator<<(const alps::alea::complex_op<T> &x)
+    {
+        *this << x.rere() << x.reim() << x.imre() << x.imim();
+        return *this;
+    }
+
+    // Store ALEA results
+    template <typename T>
+    typename std::enable_if<alps::alea::is_alea_result<T>::value, MockArchive &>::type
+    operator<<(const T &result)
+    {
+        save(*this, result, 0);
+        return *this;
+    }
+
+    // Extract simple types
+    MockArchive & operator>>(double &x) { extract_fundamental(x); return *this; }
+    MockArchive & operator>>(long &x) { extract_fundamental(x); return *this; }
+    MockArchive & operator>>(unsigned long &x) { extract_fundamental(x); return *this; }
+    MockArchive & operator>>(std::complex<double> &x) {
+      double r, i;
+      extract_fundamental(r);
+      extract_fundamental(i);
+      x = std::complex<double>(r, i);
+      return *this;
+    }
+
+    // Extract complex_op<T>
+    template<typename T>
+    MockArchive & operator>>(alps::alea::complex_op<T> &x)
+    {
+        *this >> x.rere() >> x.reim() >> x.imre() >> x.imim();
+        return *this;
+    }
+
+    // Extract ALEA results
+    template <typename T>
+    typename std::enable_if<alps::alea::is_alea_result<T>::value, MockArchive &>::type
+    operator>>(T &result)
+    {
+        load(*this, result, 0);
+        return *this;
+    }
+
+private:
+
+    template<typename T>
+    void store_fundamental(T x) {
+        unsigned char* p = reinterpret_cast<unsigned char*>(&x);
+        for(int n = 0; n < sizeof(T); ++n)
```

Comment:
It would also be good to use `std::size_t` rather than `int` here: one less warning with `-Wall`.